repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilterLoader.scala

```diff
@@ -0,0 +1,321 @@
+            }
+
+            //Only log this repose ready if we're able to properly fire up a new filter chain
+            logger.info("{} -- Repose ready", nodeId)
+            //Update the JMX bean with our status
+            configurationInformation.updateNodeStatus(nodeId, true)
+          } catch {
+            case fie: FilterInitializationException =>
+              logger.error("{} -- Unable to create new filter chain.", nodeId, fie)
+              //Update the JMX bean with our status
+              configurationInformation.updateNodeStatus(nodeId, false)
+          }
+        case _ =>
+          logger.error("{} -- Unhealthy system-model config (cannot identify local node, or no default destination) - please check your system-model.cfg.xml", nodeId)
+          healthCheckServiceProxy.reportIssue(SystemModelConfigHealthReport,
+            "Unable to identify the local host in the system model, or no default destination - please check your system-model.cfg.xml",
+            Severity.BROKEN)
+      }
+    }
+  }
+
+  @throws[FilterInitializationException]
+  private def buildFilterContexts(servletContext: ServletContext, filtersToCreate: List[FilterConfig]): List[FilterContext] = {
+    filtersToCreate.map { filterToCreate =>
+      if (classLoaderManagerService.hasFilter(filterToCreate.getName)) {
+        loadFilterContext(filterToCreate, classLoaderManagerService.getLoadedApplications, servletContext)
+      } else {
+        val message = s"Unable to satisfy requested filter chain - none of the loaded artifacts supply a filter named ${filterToCreate.getName}"
+        logger.error(message)
+        throw new FilterInitializationException(message)
+      }
+    }
+  }
+
+  /**
+    * Load a FilterContext for a filter
+    *
+    * @param filter             the Jaxb filter configuration information from the system-model
+    * @param loadedApplications The list of EarClassLoaders
+    * @return a FilterContext containing an instance of the filter and metatadata
+    * @throws org.openrepose.powerfilter.FilterInitializationException
+    */
+  @throws[FilterInitializationException]
+  private def loadFilterContext(filter: FilterConfig, loadedApplications: util.Collection[EarClassLoaderContext], servletContext: ServletContext): FilterContext = {
+    // TODO: Clean this up \/\/\/\/
+    var filterType: FilterType = null
+    var filterClassLoader: ClassLoader = null
+    val classLoaderContextIterator = loadedApplications.iterator
+    while ( {
+      classLoaderContextIterator.hasNext && filterType == null
+    }) {
+      val classLoaderContext = classLoaderContextIterator.next
+      filterType = classLoaderContext.getEarDescriptor.getRegisteredFilters.get(filter.getName)
+      if (filterType != null) filterClassLoader = classLoaderContext.getClassLoader
+    }
+    // TODO: Clean this up /\/\/\/\
+
+    // FilterType and filterClassloader are guaranteed to not be null, by a different check in the previous method
```

Comment:
this seems untrue

Reply:
Yep. That was a copy/paste of a comment from the old Java version.
Updated.